Replace SubmittableTextarea by new SmartTextarea

davidmz opened the pull request:
The internal SubmittableTextarea component replaced by new SmartTextarea. The SmartTextarea allows to:
 - submit text by Enter of Ctrl/Cmd+Enter;
 - handle files paste and drag-n-drop;
 - insert text to the cursor position using 'insertText' instance method;
 - use 'onText' attribute to handle updated text (it is necessary for 'insertText' updates, which doesn't trigger onChange).

This component handles all tasks related to creating/editing posts and comments in a unified way.

n1313 commented:
Круто! Новый компонент настолько совместим со старым, что не пришлось изменять ни одного теста?

davidmz commented:
Я думаю, это больше говорит о тестах, чем о совместимости:) Но да, он более-менее drop-in.

davidmz committed:
Replace SubmittableTextarea by new SmartTextarea

The internal SubmittableTextarea component replaced by new SmartTextarea. The SmartTextarea allows to:
 - submit text by Enter of Ctrl/Cmd+Enter;
 - handle files paste and drag-n-drop;
 - insert text to the cursor position using 'insertText' instance method;
 - use 'onText' attribute to handle updated text (it is necessary for 'insertText' updates, which doesn't trigger onChange).

This component handles all tasks related to creating/editing posts and comments in a unified way.
